pkg/client/william.go

```diff
@@ -0,0 +1,233 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+)
+
+// William Interface
+type William interface {
+	ByPass()
+	SetClient(client HttpClient)
```

Comment:
Don't expose a Set method; the idiomatic way would be to use a concrete with a public field - callers can set it as they see fit, using regular struct initialization.